File: go/cars-assemble/cars_assemble.go
// Package cars implements the [Cars Assemble] exercise.
//
// [Cars Assemble]: https://exercism.org/tracks/go/exercises/cars-assemble
package cars

import "math/bits"

// CalculateWorkingCarsPerHour calculates how many working cars are produced by
// the assembly line every hour.
func CalculateWorkingCarsPerHour(productionRate int, successRate float64) float64 {
	return float64(productionRate) * (successRate / 100)
}

// CalculateWorkingCarsPerMinute calculates how many working cars are produced
// by the assembly line every minute.
func CalculateWorkingCarsPerMinute(productionRate int, successRate float64) int {
	workingCarsPerHour := CalculateWorkingCarsPerHour(productionRate, successRate)
	return int(workingCarsPerHour / 60)
}

// CalculateCost calculates the cost of producing the given number of cars.
func CalculateCost(carsCount int) uint {
	quotient, remainder := bits.Div(0, uint(carsCount), 10)
	return (quotient * 95000) + (remainder * 10000)
}

File: clojure/cars-assemble/src/cars_assemble.clj
(ns cars-assemble)

(defn production-rate
  "Given an assembly line's `speed` (an integer between 0 and 10 inclusive),
  compute the hourly production rate, taking into account the success rate."
  [speed]
  {:pre [(<= 0 speed) (<= speed 10)]}
  (-> (cond
        (= 0 speed) 0.0
        (and (<= 1 speed) (<= speed 4)) 1.0
        (and (<= 5 speed) (<= speed 8)) 0.9
        (= 9 speed) 0.8
        (= 10 speed) 0.77)
      (* speed 221)))

(defn working-items
  "Calculate how many working cars are produced per minute."
  [speed]
  {:pre [(int? speed)]}
  (-> speed
      production-rate
      (/ 60)
      int))
